src/sortStudents.ts

```diff
@@ -1,16 +1,51 @@
-
 export interface Student {
-  // describe Student interface
+  name: string,
+  surname: string,
+  age: number,
+  married: boolean,
+  grades: number[],
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name = 'name',
+  Surname = 'surname',
+  Age = 'age',
+  Married = 'married',
+  AverageGrade = 'grades',
+}
+
+export type SortOrder = 'asc' | 'desc';
+
+function getAverageGrades(grades: number[]): number {
+  return grades.reduce((acc, grade) => acc + grade, 0) / grades.length;
 }
 
-// create SortOrder type
-export type SortOrder;
+export function sortStudents(
+  students: Student[], sortBy: SortType, order: SortOrder,
+): Student[] {
+  const copyStudents = [...students];
+
+  const compareStudents = (a: Student, b: Student): number => {
+    switch (sortBy) {
+      case SortType.Name:
+      case SortType.Surname:
+        return a[sortBy].localeCompare(b[sortBy]);
+
+      case SortType.Age:
+        return a[sortBy] - b[sortBy];
+
+      case SortType.Married:
+        return a[sortBy] && !b[sortBy] ? -1 : 1;
+
```

Comment:
```suggestion

      case SortType.Age:
      case SortType.Married:
        return +a[sortBy] - +b[sortBy];
```
we can group these cases also